src/clang-cpp-frontend/clang_cpp_convert.cpp

```diff
@@ -554,6 +555,44 @@ bool clang_cpp_convertert::get_expr(const clang::Stmt &stmt, exprt &new_expr)
     break;
   }
 
+  case clang::Stmt::CXXDeleteExprClass:
+  {
+    const clang::CXXDeleteExpr &de =
+      static_cast<const clang::CXXDeleteExpr &>(stmt);
+
+    if(de.isArrayFormAsWritten())
+      new_expr = side_effect_exprt("cpp_delete[]", empty_typet());
+    else
+      new_expr = side_effect_exprt("cpp_delete", empty_typet());
+
+    exprt arg;
+    if(get_expr(*de.getArgument(), arg))
+      return true;
+
+    new_expr.move_to_operands(arg);
+
+    // If it is a struct/union/class, get the destructor
+    if(de.getDestroyedType()->getAsCXXRecordDecl())
+    {
+      typet destt;
+      if(get_type(de.getDestroyedType(), destt))
+        return true;
+
+      // TODO: Move to adjust
```

Comment:
Can you move this to adjust? It should probably go into where the `// nothing` comment is. 

Do we have test cases for it?